github/commands/pull_request.py

```diff
@@ -94,39 +94,84 @@ def on_select_action(self, idx):
             self.open_pr_in_browser()
 
     def fetch_and_checkout_pr(self, branch_name=None):
-        self.window.status_message("Fetching PR commit...")
-        self.git(
-            "fetch",
-            self.pr["head"]["repo"]["clone_url"],
-            self.pr["head"]["ref"]
-        )
-
         if branch_name:
-            self.window.status_message("Creating local branch for PR...")
+            self.create_branch_for_pr(branch_name, checkout=True)
+        else:  # detached HEAD
+            self.checkout_detached()
+
+    def checkout_detached(self):
+        self.window.status_message("Checking out PR...")
+
+        clone_url = self.pr["head"]["repo"]["clone_url"]
+        ssh_url = self.pr["head"]["repo"]["ssh_url"]
+
+        def on_select_url(index):
+            if index < 0:
+                return
+            elif index == 0:
+                url = clone_url
+            elif index == 1:
+                url = ssh_url
+
             self.git(
-                "branch",
-                branch_name,
-                self.pr["head"]["sha"]
+                "fetch",
+                url,
+                self.pr["head"]["ref"]
             )
 
-        self.window.status_message("Checking out PR...")
-        self.checkout_ref(branch_name or self.pr["head"]["sha"])
-        util.view.refresh_gitsavvy_interfaces(self.window, refresh_sidebar=True)
+            self.checkout_ref(self.pr["head"]["sha"])
+            util.view.refresh_gitsavvy_interfaces(self.window, refresh_sidebar=True)
 
-    def create_branch_for_pr(self, branch_name):
-        self.window.status_message("Fetching PR commit...")
-        self.git(
-            "fetch",
-            self.pr["head"]["repo"]["clone_url"],
-            self.pr["head"]["ref"]
+        sublime.set_timeout(
+            lambda: self.window.show_quick_panel(
+                [clone_url, ssh_url], on_select_url)
         )
 
+    def create_branch_for_pr(self, branch_name, checkout=False):
+        if not branch_name:
+            return
+
         self.window.status_message("Creating local branch for PR...")
-        self.git(
-            "branch",
-            branch_name,
-            self.pr["head"]["sha"]
-        )
+
+        remotes = list(self.get_remotes().keys())
+        remote = self.pr["user"]["login"]
+        remote_branch = self.pr["head"]["ref"]
+
+        clone_url = self.pr["head"]["repo"]["clone_url"]
+        ssh_url = self.pr["head"]["repo"]["ssh_url"]
+
+        if remote not in remotes:
+            if not sublime.ok_cancel_dialog("Add remote '{}'?".format(remote)):
```

Comment:
Now you cannot checkout a PR without adding the remote. Is this intended?

Reply:
I don't know if it is a good default. Do you think we should proceed without setting upstream?

Reply:
Or we modify the behavior, `Checkout as local branch` sets upstream and `Create local branch` does not.